PyTAPS/connection.py

```
@@ -28,367 +30,165 @@ def __init__(self, preconnection):
+        # If security_parameters were given, initialize ssl context
+        if self.security_parameters:
+            self.security_context = ssl.create_default_context(
+                                                ssl.Purpose.SERVER_AUTH)
+            if self.security_parameters.identity:
+                print_time("Identity: " +
+                           str(self.security_parameters.identity))
+                self.security_context.load_cert_chain(
+                                        self.security_parameters.identity)
+            for cert in self.security_parameters.trustedCA:
+                self.security_context.load_verify_locations(cert)
+        # Resolve address
+        remote_info = await self.loop.getaddrinfo(
+            self.remote_endpoint.host_name, self.remote_endpoint.port)
+        self.remote_endpoint.address = remote_info[0][4][0]
+        for candidate in candidate_set:
+
+            if self.state == ConnectionState.ESTABLISHED:
+                break
+
+            if candidate[0] == 'udp':
+                self.protocol = 'udp'
+                print_time("Creating UDP connect task.", color)
+                multicast_receiver = False
+                if self.local_endpoint:
+                    if self.local_endpoint.address:
+                        print_time("local endpoint=%s" % (self.local_endpoint.address), color)
+                        check_addr = ipaddress.ip_address(self.local_endpoint.address)
+                        if check_addr.is_multicast:
+                            print_time("addr is multicast", color)
+                            if self.transport_properties.properties.get('direction') == 'unidirection-receive':
+                                print_time("direction is unicast receive", color)
+                                multicast_receiver = True
+                                self.connection = Connection(self)
+                                asyncio.create_task(self.multicast_join())
```

What if the address is multicast, but the transport property "unidirection-receive" is missing?
Should this throw an "invalid configuration" error?